core/cog-viewport.c

```diff
@@ -86,6 +109,52 @@ cog_viewport_get_property(GObject *object, unsigned prop_id, GValue *value, GPar
     }
 }
 
+/**
+ * cog_viewport_get_impl_type:
+ *
+ * Get the type of the viewport implementation in use.
+ *
+ * This function always returns a valid type. If the active
+ * [class@CogPlatform] does not provide a custom viewport type, the default
+ * built-in type included as part of the Cog core library is returned.
+ *
+ * When creating viewports, this function must be used to retrieve the
+ * type to use, e.g.:
+ *
+ * ```c
+ * CogViewport *view = g_object_new(cog_viewport_get_impl_type(), NULL);
+ * ```
+ *
+ * In most cases it should be possible to use the convenience function
+ * [ctor@Viewport.new], which uses this function internally.
+ *
+ * Returns: Type of the view implementation in use.
+ */
+GType
+cog_viewport_get_impl_type(void)
+{
+    static GType viewport_type;
+    static GOnce once = G_ONCE_INIT;
+
+    g_once(&once, (GThreadFunc) _cog_viewport_get_impl_type_init, &viewport_type);
+
+    return viewport_type;
+}
+
+/**
+ * cog_viewport_new: (constructor)
+ *
+ * Creates a new instance of a viewport implementation and sets its properties.
+ *
+ * Returns: (transfer full): A new web view.
+ */
+CogViewport *
+cog_viewport_new()
+{
+    void *viewport = g_object_new(cog_viewport_get_impl_type(), NULL);
+    return viewport;
```

Comment:
This can be written in a single line:

```suggestion
    return g_object_new(cog_viewport_get_impl_type(), NULL);
```

Reply:
fixed